### PR TITLE
Always set response.URL to the latest redirect even on errors fix #987

### LINKED ISSUE #987: Redirect Error Reporting
In v0.23.0 a redirect to a down server provided an error including the failing URL like:

error="Post http://127.0.0.1?session_state=29de8415-ea47-4919-a640-4db59cbb953e&code=02e7e7c9-56e7-4c43-a709-af145d4fe5d8.29de8415-ea47-4919-a640-4db59cbb953e.2d78ebc3-9903-4753-a00d-28c051498a49: dial tcp 127.0.0.1:80: connect: connection refused"

In v0.24.0 while the redirect URL still shows on the console output in a WARN entry, the failing redirect URL is no longer available in the error attribute of the response.  The error looks like:

error="dial tcp 127.0.0.1:80: connect: connection refused"

In v0.24.0 how can I programmatically determine which URL had the failure?

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
I am not particularly proud of this fix(changing values from inside lambdas is not great Idea IMO) but I couldn't figure out anything else that won't require some serious rewrite. 
Also I found we don't correctly catch connection refused on dial but the fix broke a lot of tests so I decided to put it in another commit/PR to not pollute this one with too many unrelated changes

File: js/modules/k6/http/request_test.go [hunk 1]
```diff
@@ -1348,6 +1348,11 @@ func TestErrorCodes(t *testing.T) {
 	defer tb.Cleanup()
 	sr := tb.Replacer.Replace
 
+	var connectionRefusedErrorText = "connect: connection refused"
+	if runtime.GOOS == "windows" {
+		connectionRefusedErrorText = "connectex: No connection could be made because the target machine actively refused it."
```
Comment (robingustafsson) [inline js/modules/k6/http/request_test.go]: Why is the "No" captialized but the "connection" (above) not?

Comment (MStoykov) [reply js/modules/k6/http/request_test.go]: This is what windows/golang gives us back ... and as far as I can find out this windows text comes from outside go ... so :(

Comment (robingustafsson) [reply js/modules/k6/http/request_test.go]: Ah, missed that it was in a test case actually, was in the mindset that it was a string we set for `res.error` 😄

**Comment by MStoykov** [comment]
@na-- that code is if we don't have an error and we have a response.
 In this particular case we don't have a response at that time as we couldn't make the last request. The only place where we have the url is before we actually make it after we have gotten `302`.
 I would prefer if we rewrite this so that the redirect is more explicit and get rid of the lambda but this will probably not help with the amount of code or change it all that much it will just not be in the lambda ;). 

As far as errors goes - If you look carefully I check that the `res.url` we get is what I expect and those errors are inside the script not outside and are after the check on whether the `res.url` has the correct value

**Pull request merged**